### ISSUE TITLE
implicit_multiplication_application transformation in parse_expr incorrectly handles method

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#21020]
Example

```py
>>> from sympy.parsing.sympy_parser import standard_transformations,\
... implicit_multiplication_application, parse_expr
>>> transformations = (standard_transformations +
...     (implicit_multiplication_application,))
>>> parse_expr('(exp(x) / (1 + exp(2x))).subs(exp(x), u)', transformations=transformations)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "./sympy/parsing/sympy_parser.py", line 1008, in parse_expr
    return eval_expr(code, local_dict, global_dict)
  File "./sympy/parsing/sympy_parser.py", line 903, in eval_expr
    code, global_dict, local_dict)  # take local objects in preference
  File "<string>", line 1, in <module>
TypeError: can't multiply sequence by non-int of type 'method'
```

Putting a print statement in `eval_expr` (side note, parse_expr ought to have better error reporting so this isn't necessary), it is trying to evaluate the expression

```
(exp (Symbol ('x' ))/(Integer (1 )+exp (Integer (2 )*Symbol ('x' )))).subs *(exp (Symbol ('x' )),Symbol ('u' ))
```

This used to work. 

This is an example on SymPy Gamma. See https://github.com/sympy/sympy_gamma/issues/172.

**Comment by asmeurer** [comment]
Actually I'm not sure that it ever used to work. I tried SymPy 0.7.5, which is the version that used to be on Gamma, and it still fails (even in Python 2). Maybe this example was just never tested? Or something else changed in Gamma to make it stop working. CC @aktech 

Regardless, we should fix this. It should recognize that a dotted name is not something that should be multiplied implicitly.

**Comment by asmeurer** [comment]
Fix at #21021. I also included a fix that makes the error message better in cases like this.
